Pd 1163 expand local build contributing content (#2909)
truenas/documentation: content/Contributing/Documentation/ContentUpdate.md

Remember that this snippet file is used in multiple documents, so only content that generally applies is included in the snippet.
Changes that are specific to a certain piece of content are added directly to that article.

> ## Differences in Master and Version Branches

Reviewer: ```suggestion
##  Master and Version Branch Differences
```

> Due to numerous software updates and version releases, the ***24.04*** and ***23.10*** SCALE branches have been created to assist in the organization of the Documentation Hub. 

Reviewer: I'm not a fond of this sentence at all, Maybe this instead:
```suggestion
TrueNAS documentation uses branches to organize content related to specific software releases and updates, for example the ***24.04*** and ***23.10*** SCALE branches of the Documentation Hub. 
```

> + Changes to content within the ***master*** branch result in changes to the TrueNAS SCALE Nightly Development Documentation content. To ensure your edits are submitted to the ***master*** branch, create a simple copy ("fork") of the main repository and check that ***master*** is listed as the *base*.

Reviewer: I think we should make this more active voice. Suggestion:
```suggestion
+ Content changes in the ***master*** branch result in changes to the TrueNAS SCALE Nightly Development Documentation content. To make changes in the ***master*** branch, create a simple copy ("fork") of the main repository, select that forked *master* as the *base* for your changes. Do not make changes directly to the **master** branch of the Documentation Hub repository.
```

+ Changes to material in the ***24.04*** branch result in modifications to the TrueNAS SCALE 24.04 (Dragonfish) Documentation content. To ensure your edits are proposed to the ***24.04*** branch, create a fork of the main repository and check that ***24.04*** is listed as the *base*.

> + Changes to content within the ***23.10*** branch cause changes to the TrueNAS SCALE 23.10 (Cobia) Documentation content. To ensure your edits are submitted to the ***23.10*** branch, create a fork of the main repository and check that ***23.10*** is listed as the *base*.

Reviewer: I think these can be more active voice as well. Make definitive active voice statements here. Users reading this need specific directions about what they need to do or they get confused and will do the wrong thing. If they need to fork each of these branches say that as the first thing. Like "To make changes in the 23.10 branch, create a fork of that branch, make your changes and submit your PR." or whatever the directions are.

> ## Forking the Repo

Reviewer: ```suggestion
## Fork the Repo
```

> To submit a change, you'll create a fork of the main repository, edit your copy, then propose "merging" your changes back into the main repository.

Reviewer: do not use contractions as ESL users have trouble with translating these words. Also do not use " " unless you are quoting someone. If you want to say create a PR to propose merging your changes that is a good alternative to the double quotes.
```suggestion
To submit a change, create a fork of the main repository, edit your copy, then propose merging your changes back into the main repository.
```
